knowledge-base/grid-sort-descending.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Sort Grid Column Descending
+description: How to sort a Grid column descending first, and then ascending.
+type: how-to
+page_title: How to Sort a Grid Column Descending First
+slug: grid-kb-sort-descending
+position: 
+tags: grid, sorting, state
+ticketid:
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+How to sort a Grid column descending first, and then ascending?
+
+How to reverse the Grid sorting logic for a column?
+
+
+## Solution
+
+This scenario requires knowledge about the [Grid State]({%slug grid-state%}). Get familiar with the following sections first:
+
+* [Information in the Grid State]({%slug grid-state%}#information-in-the-grid-state)
+* [Grid `OnStateChanged` Event]({%slug grid-state%}#onstatechanged)
+* [Grid State Methods]({%slug grid-state%}#methods)
+
+The Grid always tries to maintain the following order of sorting states for each column:
+
+No sorting &gt; Ascending &gt; Descending &gt; No sorting &gt; ...
+
+Let's assume that the Grid has a `Stock` column which is not sorted, but it should be sorted descending first, if the user clicks on the header. The required algorithm to interept sorting and change the sort direction is:
+
+1. Set a helper variable that will keep the previous sort state of the `Stock` column. If the column is unsorted by default, then the helper variable can be equal to `null`.
+1. Subscribe to the [Grid `OnStateChanged` Event]({%slug grid-state%}#onstatechanged).
+1. Check if the user has changed the sort state by checking if `args.PropertyName` is `"SortDescriptors"`.
+1. If yes, then iterate `args.GridState.SortDescriptors` and check if the `Stock` column is now sorted, what is the sort direction, and what was the prevous sort direction.
+1. Depending on the current situation, either switch the `SortDirection` property of the [`SortDescriptor`](/blazor-ui/api/telerik.datasource.sortdescriptor), or add a new `SortDescriptor` to the `args.GridState.SortDescriptors`.
+1. Use the [Grid `SetStateAsync` method]({%slug grid-state%}#setstateasync) to apply the modified Grid state to the component instance.
+
+>caption Sort a Grid column descending first
+
+````CSHTML
+@using Telerik.DataSource
```

Comment:
I think the sample should be refactored, I am experiencing some odd behavior. First click on the `Stock` column header indeed sorts descending. However, a click on the `Name` column header afterwards modifies the sort direction of the `Stock column.

Reply:
Agree. Done.